src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -71,6 +71,42 @@ public List<TElement> ToList()
             return list;
         }
 
+        public HashSet<TElement> ToHashSet(IEqualityComparer<TElement> comparer) => new HashSet<TElement>(_source, comparer);
+
+        public HashSet<TElement> ToHashSet(int minIdx, int maxIdx, IEqualityComparer<TElement> comparer)
```

Comment:
Nit: Make this `internal` and put this below the corresponding `ToArray` / `ToList` overloads that accept indices.